From 21b64cda6c2cd29c287c08c85962609f8d93d18f Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:28:46 -0400
Subject: [PATCH 2/6] WRS: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/openstack-aodh.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/openstack-aodh.spec b/SPECS/openstack-aodh.spec
index 9fffb3c..01615e7 100644
--- a/SPECS/openstack-aodh.spec
+++ b/SPECS/openstack-aodh.spec
@@ -4,7 +4,7 @@

 Name:             openstack-aodh
 Version:          5.1.0
-Release:          1%{?dist}
+Release:          1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:          OpenStack Telemetry Alarming
 License:          ASL 2.0
 URL:              https://github.com/openstack/aodh.git
--
1.8.3.1

