From 8620101244ea5be1ff0cc0e127fa57dca4be468a Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:28:46 -0400
Subject: [PATCH] WRS: spec-include-TiS-patches.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/openstack-aodh.spec | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/SPECS/openstack-aodh.spec b/SPECS/openstack-aodh.spec
index 01615e7..f840861 100644
--- a/SPECS/openstack-aodh.spec
+++ b/SPECS/openstack-aodh.spec
@@ -18,6 +18,9 @@ Source12:         %{name}-notifier.service
 Source13:         %{name}-expirer.service
 Source14:         %{name}-listener.service

+#WRS: Include patches here:
+Patch1:           0001-modify-aodh-api.patch
+Patch2:           0002-Add-drivername-support-for-postgresql-connection-set.patch
 BuildArch:        noarch

 BuildRequires:    openstack-macros
@@ -212,6 +215,10 @@ This package contains the Aodh test files.
 %prep
 %setup -q -n %{pypi_name}-%{upstream_version}

+#WRS: Apply patches here
+%patch1 -p1
+%patch2 -p1
+
 find . \( -name .gitignore -o -name .placeholder \) -delete

 find aodh -name \*.py -exec sed -i '/\/usr\/bin\/env python/{d;q}' {} +
@@ -259,10 +266,12 @@ install -p -D -m 640 %{SOURCE1} %{buildroot}%{_datadir}/aodh/aodh-dist.conf
 install -p -D -m 640 aodh/aodh.conf %{buildroot}%{_sysconfdir}/aodh/aodh.conf
 install -p -D -m 640 aodh/api/policy.json %{buildroot}%{_sysconfdir}/aodh/policy.json

+#WRS
+install -p -D -m 640 aodh/api/aodh-api.py %{buildroot}%{_datadir}/aodh/aodh-api.py
 # Setup directories
 install -d -m 755 %{buildroot}%{_sharedstatedir}/aodh
 install -d -m 755 %{buildroot}%{_sharedstatedir}/aodh/tmp
-install -d -m 750 %{buildroot}%{_localstatedir}/log/aodh
+install -d -m 755 %{buildroot}%{_localstatedir}/log/aodh

 # Install logrotate
 install -p -D -m 644 %{SOURCE2} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
@@ -340,12 +349,12 @@ exit 0
 %files common -f %{pypi_name}.lang
 %doc README.rst
 %dir %{_sysconfdir}/aodh
+%{_datadir}/aodh/aodh-api.*
 %attr(-, root, aodh) %{_datadir}/aodh/aodh-dist.conf
 %config(noreplace) %attr(-, root, aodh) %{_sysconfdir}/aodh/aodh.conf
 %config(noreplace) %attr(-, root, aodh) %{_sysconfdir}/aodh/policy.json
 %config(noreplace) %{_sysconfdir}/logrotate.d/%{name}
-%dir %attr(0750, aodh, root)  %{_localstatedir}/log/aodh
-%{_bindir}/aodh-dbsync
+%dir %attr(0755, aodh, root)  %{_localstatedir}/log/aodh
 %{_bindir}/aodh-config-generator

 %defattr(-, aodh, aodh, -)
@@ -353,6 +362,7 @@ exit 0
 %dir %{_sharedstatedir}/aodh/tmp

 %files api
+%{_bindir}/aodh-dbsync
 %{_bindir}/aodh-api
 %{_unitdir}/%{name}-api.service

--
2.7.4

