From 337ef0a6bdaa310f008f58a544c7d5e48dbcf48a Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 26 Sep 2016 17:32:55 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/python-wsme.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-wsme.spec b/SPECS/python-wsme.spec
index 2d03df4..ccecfc2 100644
--- a/SPECS/python-wsme.spec
+++ b/SPECS/python-wsme.spec
@@ -14,7 +14,7 @@
 
 Name:           python-%{lpypi_name}
 Version:        0.9.2
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Web Services Made Easy
 
 License:        MIT
-- 
2.7.4

