From 6aa8e5b13954e7b2ef2ffe619c8818db4a109f6b Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Tue, 6 Mar 2018 09:56:17 -0600
Subject: Remove TOX calls from build

---
 SPECS/python-wsme.spec | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/SPECS/python-wsme.spec b/SPECS/python-wsme.spec
index c778f97..403c331 100644
--- a/SPECS/python-wsme.spec
+++ b/SPECS/python-wsme.spec
@@ -102,12 +102,6 @@ manipulate the request and the response objects.
 %{__python3} setup.py install --skip-build --root %{buildroot}
 %endif
 
-%check
-%{__python2} setup.py test
-%if 0%{?with_python3}
-%{__python3} setup.py test
-%endif
-
 %files -n python2-%{lpypi_name}
 %doc README.rst examples/
 %license LICENSE
-- 
2.7.4

