From 6ded6aa6c81808a716628196caa91f20545e2cc0 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Tue, 3 Jan 2017 13:36:15 -0500
Subject: [PATCH] meta patch for ClientSideError logging verbosity fix.

---
 SPECS/python-wsme.spec | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SPECS/python-wsme.spec b/SPECS/python-wsme.spec
index 2f84ba5..2d03df4 100644
--- a/SPECS/python-wsme.spec
+++ b/SPECS/python-wsme.spec
@@ -20,6 +20,10 @@ Summary:        Web Services Made Easy
 License:        MIT
 URL:            https://pypi.python.org/pypi/WSME
 Source0:        http://tarballs.openstack.org/wsme/%{pypi_name}-%{version}.tar.gz
+
+# WRS
+Patch0001: 0001-log-client-side-errors.patch
+
 BuildArch:      noarch
 
 %description
@@ -83,6 +87,9 @@ manipulate the request and the response objects.
 %prep
 %setup -q -n %{pypi_name}-%{upstream_version}
 
+# Apply WRS patches
+%patch0001 -p1
+
 %build
 %{__python2} setup.py build
 %if 0%{?with_python3}
-- 
2.7.4

