From 7feac57d571e49e042adb96738a3688c56adade0 Mon Sep 17 00:00:00 2001
From: Tyler Smith <tyler.smith@windriver.com>
Date: Mon, 8 Apr 2019 15:33:16 -0400
Subject: [PATCH 1/1] Add keyring patch

---
 SPECS/openstack-keystone.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/openstack-keystone.spec b/SPECS/openstack-keystone.spec
index a20bda1..945de6d 100644
--- a/SPECS/openstack-keystone.spec
+++ b/SPECS/openstack-keystone.spec
@@ -28,7 +28,7 @@ Name:           openstack-keystone
 # https://review.openstack.org/#/q/I6a35fa0dda798fad93b804d00a46af80f08d475c,n,z
 Epoch:          1
 Version:        15.0.0
-Release:        0.2%{?milestone}%{?dist}
+Release:        0.2.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        OpenStack Identity Service
 License:        ASL 2.0
 URL:            http://keystone.openstack.org/
@@ -42,6 +42,9 @@ Source3:        openstack-keystone.sysctl
 Source5:        openstack-keystone-sample-data
 Source20:       keystone-dist.conf

+# STX: Include patches here
+Patch1:         0001-Rebasing-Keyring-integration.patch
+
 BuildArch:      noarch
 BuildRequires:  openstack-macros
 BuildRequires:  python%{pyver}-devel
--
1.8.3.1
