From e3f1154184f24103942c98fa9373763df49703c7 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:32:49 -0400
Subject: WRS: 0002-Packages-sdk-for-remote-clients.patch

---
 SPECS/python-heatclient.spec | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/SPECS/python-heatclient.spec b/SPECS/python-heatclient.spec
index c5ff419..1f44763 100644
--- a/SPECS/python-heatclient.spec
+++ b/SPECS/python-heatclient.spec
@@ -92,6 +92,13 @@ the OpenStack Heat API.
 
 This package contains auto-generated documentation.
 
+%package          sdk
+Summary:          SDK files for %{name}
+
+%description      sdk
+Contains SDK files for %{name} package
+
+
 %prep
 %autosetup -n %{name}-%{upstream_version} -S git
 
@@ -128,6 +135,10 @@ install -pm 644 tools/heat.bash_completion \
 # Delete tests
 rm -fr %{buildroot}%{python2_sitelib}/heatclient/tests
 
+# prep SDK package
+mkdir -p %{buildroot}/usr/share/remote-clients/%{name}
+tar zcf %{buildroot}/usr/share/remote-clients/%{name}/%{name}-%{version}.tgz --exclude='.gitignore' --exclude='.gitreview' -C .. %{name}-%{version}
+
 
 %{__python2} setup.py build_sphinx -b html
 # Fix hidden-file-or-dir warnings
@@ -164,6 +175,9 @@ install -p -D -m 644 doc/build/man/heat.1 %{buildroot}%{_mandir}/man1/heat.1
 %doc doc/build/html
 %license LICENSE
 
+%files sdk
+/usr/share/remote-clients/%{name}/%{name}-%{version}.tgz
+
 %changelog
 * Mon Aug 14 2017 Alfredo Moralejo <amoralej@redhat.com> 1.11.0-1
 - Update to 1.11.0
-- 
2.7.4

