diff --git a/aodhclient/osc.py b/aodhclient/osc.py
index 1e8599f..a1ecfee 100644
--- a/aodhclient/osc.py
+++ b/aodhclient/osc.py
@@ -39,7 +39,9 @@ def make_client(instance):
         API_VERSIONS)
     # NOTE(sileht): ensure setup of the session is done
     instance.setup_auth()
-    return aodh_client(session=instance.session)
+    return aodh_client(session=instance.session,
+                       interface=instance.interface,
+                       region_name=instance.region_name)


 def build_option_parser(parser):
