From c047aecf35d067390d0e80e0149725c3031a19c3 Mon Sep 17 00:00:00 2001
From: Tyler Smith <tyler.smith@windriver.com>
Date: Fri, 29 Jun 2018 20:07:09 -0500
Subject: [PATCH 1/1] [PATCH] meta patch for distributed keystone fix

---
 SPECS/python-django-openstack-auth.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/python-django-openstack-auth.spec b/SPECS/python-django-openstack-auth.spec
index 90b4692..89acf21 100644
--- a/SPECS/python-django-openstack-auth.spec
+++ b/SPECS/python-django-openstack-auth.spec
@@ -19,6 +19,7 @@ Patch0001: 0001-Pike-rebase-for-openstack-auth.patch
 Patch0002: 0002-disable-token-validation-per-auth-request.patch
 Patch0003: 0003-cache-authorized-tenants-in-cookie-to-improve-performance.patch
 Patch0004: 0004-Distributed-Keystone.patch
+Patch0005: fix_for_dc_region_switching.patch

 BuildArch:      noarch
 
-- 
2.7.4

