From a31e3b402f19767e1390edf17400e38f6592c829 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Thu, 7 Dec 2017 09:12:43 -0500
Subject: [PATCH] Disable check on build


diff --git a/SPECS/python-oslo-messaging.spec b/SPECS/python-oslo-messaging.spec
index 627a1be..648718a 100644
--- a/SPECS/python-oslo-messaging.spec
+++ b/SPECS/python-oslo-messaging.spec
@@ -268,11 +268,11 @@ rm -fr doc/build/html/.buildinfo
 %check
 # Temporarily disabling tests until we have
 # mock >= 1.2 and pika_pool
-%{__python2} setup.py test ||
-%if 0%{?with_python3}
-rm -rf .testrepository
-%{__python3} setup.py test ||
-%endif
+# %{__python2} setup.py test ||
+# %if 0%{?with_python3}
+# rm -rf .testrepository
+# %{__python3} setup.py test ||
+# %endif
 
 %files -n python2-%{pkg_name}
 %license LICENSE
