diff --git a/SPECS/python-openstackclient.spec b/SPECS/python-openstackclient.spec
index 3ef7f0d..9f624f1 100644
--- a/SPECS/python-openstackclient.spec
+++ b/SPECS/python-openstackclient.spec
@@ -19,6 +19,7 @@ Source0:          https://tarballs.openstack.org/%{name}/%{name}-%{upstream_vers
 
 # WRS patches
 Patch0001:        0001-neutron-extensions.patch
+Patch0002:        openstackClient_Passwordchange_warning.patch 
 
 BuildArch:        noarch
 
