From bf24d1a99d29ab66feb4b2c7c7e5839e1cc23ec3 Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Thu, 2 Nov 2017 10:45:40 -0400
Subject: CGTS-7947: add --os-keystone-region-name to identity client

Signed-off-by: Andy Ning <andy.ning@windriver.com>
---
 SPECS/python-openstackclient.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/python-openstackclient.spec b/SPECS/python-openstackclient.spec
index 4791547..5d3d7b4 100644
--- a/SPECS/python-openstackclient.spec
+++ b/SPECS/python-openstackclient.spec
@@ -21,6 +21,7 @@ Source0:          https://tarballs.openstack.org/%{name}/%{name}-%{upstream_vers
 Patch0001:        0001-neutron-extensions.patch
 Patch0002:        openstackClient_Passwordchange_warning.patch 
 Patch0003:        CGTS-7814-warning-only-when-the-admin-password-chang.patch
+Patch0004:        0001-CGTS-7947-add-os-keystone-region-name-to-identity-cl.patch
 
 BuildArch:        noarch
 
-- 
2.7.4

