From 15a676bb74150546db8425ecae4976fb773661a0 Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Tue, 25 Sep 2018 14:57:06 -0400
Subject: [PATCH 16/16] Allow setting wrs-binding:vif_model using osclient

---
 SPECS/python-openstackclient.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/python-openstackclient.spec b/SPECS/python-openstackclient.spec
index c192b2b..29fec53 100644
--- a/SPECS/python-openstackclient.spec
+++ b/SPECS/python-openstackclient.spec
@@ -29,6 +29,7 @@ Patch0008:        0002-US106901-Openstack-CLI-Adoption.patch
 Patch0009:        0001-Optimize-getting-endpoint-list.patch
 Patch0010:        0002-Add-support-for-endpoing-filter-commands.patch
 Patch0011:        0003-Add-support-for-endpoint-group-commands.patch
+Patch0012:        0001-Allow-setting-wrs-binding-vif_model-using-osclient.patch

 BuildArch:        noarch

--
1.8.3.1
