From ab8fc2b85ab7b60bdfeca496a32e90bc8f575478 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Fri, 11 May 2018 13:13:15 -0400
Subject: [PATCH] meta patch for endpoint groups

Signed-off-by: Kam Nasim <kam.nasim@windriver.com>
---
 SPECS/python-openstackclient.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/python-openstackclient.spec b/SPECS/python-openstackclient.spec
index 5d75107..26941e4 100644
--- a/SPECS/python-openstackclient.spec
+++ b/SPECS/python-openstackclient.spec
@@ -27,6 +27,9 @@ Patch0006:        0002-US101470-Openstackclient-implementation-of-novaclien.patc
 Patch0007:        0001-US106901-Openstack-CLI-Adoption.patch
 Patch0008:        0002-US106901-Openstack-CLI-Adoption.patch
 Patch0009:        0003-US106901-Openstack-CLI-Adoption.patch
+Patch0010:        0001-Optimize-getting-endpoint-list.patch
+Patch0011:        0002-Add-support-for-endpoing-filter-commands.patch
+Patch0012:        0003-Add-support-for-endpoint-group-commands.patch
 
 BuildArch:        noarch
 
-- 
1.8.3.1

