From 6cf8ebc9e09035d49cfb051c4c4b03c4b8607ba3 Mon Sep 17 00:00:00 2001
From: Teresa Ho <teresa.ho@windriver.com>
Date: Tue, 30 Oct 2018 11:16:53 -0400
Subject: [PATCH] Remove wrs-binding:mac_filtering and setting extension

---
 SPECS/python-openstackclient.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/python-openstackclient.spec b/SPECS/python-openstackclient.spec
index 29fec53..3dd52cb 100644
--- a/SPECS/python-openstackclient.spec
+++ b/SPECS/python-openstackclient.spec
@@ -30,6 +30,7 @@ Patch0009:        0001-Optimize-getting-endpoint-list.patch
 Patch0010:        0002-Add-support-for-endpoing-filter-commands.patch
 Patch0011:        0003-Add-support-for-endpoint-group-commands.patch
 Patch0012:        0001-Allow-setting-wrs-binding-vif_model-using-osclient.patch
+Patch0013:        0001-Remove-wrs-binding-mac_filtering-and-setting-extensi.patch
 
 BuildArch:        noarch
 
-- 
1.8.3.1

