From 89c5d60116569e1446c62d652d30eb0a21130193 Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Thu, 2 Nov 2017 18:26:56 +0200
Subject: [PATCH 1/2] Update package versioning for TIS format

---
 SPECS/python-glance-store.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-glance-store.spec b/SPECS/python-glance-store.spec
index 831aba4..387a326 100644
--- a/SPECS/python-glance-store.spec
+++ b/SPECS/python-glance-store.spec
@@ -7,7 +7,7 @@
 
 Name:           python-glance-store
 Version:        0.22.0
-Release:        1%{?dist}
+Release:        1%{?_tis_dist}.%{tis_patch_ver}
 Summary:        OpenStack Image Service Store Library
 
 License:        ASL 2.0
-- 
2.7.4

