From e6daf4d7dbe603e82a267d6d99a454453b902f68 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:42:44 -0400
Subject: [PATCH] WRS: spec-loopingcall-permit-aborting-while-sleeping.patch

---
 SPECS/python-oslo-service.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/python-oslo-service.spec b/SPECS/python-oslo-service.spec
index 658bb42..5ff8f34 100644
--- a/SPECS/python-oslo-service.spec
+++ b/SPECS/python-oslo-service.spec
@@ -14,6 +14,10 @@ Summary:        Oslo service library
 License:        ASL 2.0
 URL:            http://launchpad.net/oslo
 Source0:        https://tarballs.openstack.org/%{pypi_name}/%{pypi_name}-%{upstream_version}.tar.gz
+
+# WRS
+Patch0001:      loopingcall-permit-aborting-while-sleeping.patch
+
 BuildArch:      noarch
 
 %package -n     python2-%{pname}
-- 
2.7.4

